main_test.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestGetHealthCheckURL(t *testing.T) {
+	// ----- conditions -------------------------------------------------------
+	testCases := []struct {
+		name            string
+		endpoint        string
+		healthCheckPath string
+		healthCheckPort int
+		want            string
+	}{
+		{
+			name:            "PortSetToZero",
+			endpoint:        "http://minio1:9000",
+			healthCheckPath: "/minio/health/ready",
+			want:            "http://minio1:9000/minio/health/ready",
+		},
+		{
+			name:            "PortSetToNonZeroValue",
+			endpoint:        "http://minio1:9000",
+			healthCheckPath: "/minio/health/ready",
+			healthCheckPort: 4242,
+			want:            "http://minio1:4242/minio/health/ready",
+		},
```

Comment:
You could add test cases for negative aspects as well such as incorrect healthCheckPort and also incorrect endpoint - incase of latter perhaps the code might crash.

Reply:
Separated tests for valid and invalid use cases.